met-web/src/components/layout/SideNav/types.ts

```diff
@@ -1,6 +1,25 @@
+import { IconDefinition } from '@fortawesome/fontawesome-common-types';
+
 export interface SideNavProps {
     open: boolean;
     isMediumScreen: boolean;
     drawerWidth: number;
     setOpen: (open: boolean) => void;
 }
+
+export interface DrawerBoxProps {
+    isMediumScreen: boolean;
+    setOpen: (open: boolean) => void;
+}
+
+export interface IconAssignments {
```

Comment:
This seems like a lot of trouble to go to... I'm thinking you should just extend the `interface Route` to include an `icon?: IconDefinition` which you can then render in your list mapping (or the broken link otherwise). Let me know if there'd be an issue with that

Reply:
Also, aren't these other strings redundant if you have `[k: string]`?

Reply:
I changed this based on your second comment, I didn't realize that I could just define the structure and not the keys.